types/hapi/index.d.ts

```diff
@@ -571,10 +571,10 @@ export interface ServerEventCriteria {
  * Related to [See docs](https://hapijs.com/api/16.1.1#servermethodname-method-options)
  */
 export interface ServerMethod {
-    /** Not possible to improve this typing due to this unresolvable issue:  https://github.com/Microsoft/TypeScript/issues/15190 */
-    (...args: (any | ServerMethodNext)[]): void;
     /** the method must return a value (result, Error, or a promise) or throw an Error. */
     (...args: any[]): any | Error | Promise<any>;
+    /** Not possible to improve this typing due to this unresolvable issue:  https://github.com/Microsoft/TypeScript/issues/15190 */
+    (...args: (any | ServerMethodNext)[]): void;
```

Comment:
Moving the overload below the more general one means it will never be matched. the overload was wrong anyways, so just remove it all together.

the return type of the previous overload should be `any` and not `any | Error | Promise<void>`